blockwatcher: Remove extraneous log statements when there are no blocks to catch up on

```
Mar 01 15:32:42 beefcake livepeer[17450]: I0301 15:32:42.745240   17450 block_watcher.go:318] Backfilling block events (this can take a while)...
Mar 01 15:32:42 beefcake livepeer[17450]: I0301 15:32:42.745252   17450 block_watcher.go:319] Start block: 11953162                 End block: 11953161                Blocks elapsed: -1
Mar 01 15:32:42 beefcake livepeer[17450]: I0301 15:32:42.747350   17450 block_watcher.go:537] fetching block logs from=11953162 to=11953161
```

If the endBlock is before the startBlock indicating there are no blocks to catch up on, the extra log statement can be avoided to avoid user confusion.

I don't know if this is related, but also seeing this on `0.5.14` is there a connection or is this a separate issue?:
```
I0302 13:14:52.237208    8901 block_watcher.go:318] Backfilling block events (this can take a while)...
I0302 13:14:52.237235    8901 block_watcher.go:319] Start block: 11958998          End block: 11958997             Blocks elapsed: -1
I0302 13:14:52.242326    8901 block_watcher.go:537] fetching block logs from=11958998 to=11958997
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x10c48c5]

goroutine 1 [running]:
net/url.(*URL).Hostname(...)
        /usr/local/go/src/net/url/url.go:1112
main.getServiceURI(0xc0000dc160, 0x0, 0x0, 0x0, 0x0, 0x0)
        /build/cmd/livepeer/livepeer.go:1070 +0x2e5
main.main()
        /build/cmd/livepeer/livepeer.go:873 +0x5185
ubuntu@ip-172-31-33-16:~$ ./livepeer-linux-amd64/livepeer -version
Livepeer Node Version: 0.5.14-4df24536                                             Golang runtime version: gc go1.15.5                                                Architecture: amd64
Operating system: linux
ubuntu@ip-172-31-33-16:~$
```

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -8,6 +8,8 @@
 
 - \#1810 Display "n/a" in CLI when max gas price isn't specified (@kyriediculous)
 - \#1827 Limit the maximum size of a segment read over HTTP (@jailuthra)
+- \#1809 Don't log statement that blocks have been backfilled when no blocks have elapsed
+- \#1809 Avoid nil pointer error in SyncToLatestBlock when no blocks are present in the database (@kyriediculous)
 
 #### Orchestrator
 

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -8,7 +8,7 @@
 
 - \#1810 Display "n/a" in CLI when max gas price isn't specified (@kyriediculous)
 - \#1827 Limit the maximum size of a segment read over HTTP (@jailuthra)
-- \#1809 Don't log statement that blocks have been backfilled when no blocks have elapsed
+- \#1809 Don't log statement that blocks have been backfilled when no blocks have elapsed (@kyriediculous)
 - \#1809 Avoid nil pointer error in SyncToLatestBlock when no blocks are present in the database (@kyriediculous)
 
 #### Orchestrator

--- a/eth/blockwatch/block_watcher.go
+++ b/eth/blockwatch/block_watcher.go
@@ -159,6 +159,10 @@ func (w *Watcher) syncToLatestBlock() error {
 		return err
 	}
 
+	if lastSeenHeader == nil {
+		return w.pollNextBlock()
+	}
+
 	for i := lastSeenHeader.Number; i.Cmp(newestHeader.Number) < 0; i = i.Add(i, big.NewInt(1)) {
 		if err := w.pollNextBlock(); err != nil {
 			return err
@@ -332,7 +336,7 @@ func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, erro
 		return events, nil
 	}
 
-	if blocksElapsed = latestBlockNum - startBlockNum; blocksElapsed == 0 {
+	if blocksElapsed = latestBlockNum - startBlockNum; blocksElapsed <= 0 {
 		return events, nil
 	}
 

